pkg/apic/service.go

```diff
@@ -31,12 +33,13 @@ func (c *ServiceClient) PublishService(serviceBody ServiceBody) (*v1alpha1.APISe
 	if err != nil {
 		return nil, err
 	}
-	// Update description title after creating APIService to inlcude the stage name if it exists
-	c.postAPIServiceUpdate(&serviceBody)
+
 	err = c.processRevision(&serviceBody)
 	if err != nil {
 		return nil, err
 	}
+	// Update APIServiceRevision title after creating APIService to inlcude the APIServiceRevision Pattern
+	c.updateAPIServiceRevisionTitle(&serviceBody)
```

Comment:
I think this should be further up before processing the revision, since the revision will be created with NameToPush setup by agent implementation.